### PR TITLE
wgengine/magicsock, health, ipn/ipnstate: track DERP-advertised health

### PR DESCRIPTION
**Pull request by bradfitz**
And add health check errors to ipnstate.Status (tailscale status --json).

Updates #2746
Updates #2775

### CODE REVIEW

File: ipn/ipnstate/ipnstate.go [hunk 1]
```diff
@@ -38,6 +38,11 @@ type Status struct {
 	TailscaleIPs []netaddr.IP // Tailscale IP(s) assigned to this node
 	Self         *PeerStatus
 
+	// Health contains health check problems.
+	// Empty means everything is good. (or at least that no known
+	// problems are detected)
+	Health []string
```
Comment (crawshaw) [inline ipn/ipnstate/ipnstate.go:44]: Not parsable by frontends in this form.

If someday we want to report health to particular network nodes, and render it nicely in the UI next to the node, we'll want something more here. Should we pre-empt it and make a `type Health struct { Problem []string }` to add fields to?

**Review by crawshaw — APPROVED** [review]

File: ipn/ipnstate/ipnstate.go (hunk 1 above)
Comment (bradfitz) [reply ipn/ipnstate/ipnstate.go:44]: for per-peer health, we can put that on the Peer struct itself, no?

Comment (bradfitz) [reply ipn/ipnstate/ipnstate.go:44]: that is, on `Peer map[key.Public]*PeerStatus`

Comment (crawshaw) [reply ipn/ipnstate/ipnstate.go:44]: sounds good

**Pull request merged**